### PR TITLE
Make functions deploys a bit safer.

### PR DESCRIPTION
**Pull request by inlined**
1. Function deletes come after function creates, which will help make function renaming safer
2. We now explicitly do not support making a scheduled pub/sub function a generic pub/sub function (which was probably broken anyway)

### CODE REVIEW

**Review by joehan — CHANGES_REQUESTED** [review]
It looks like you didn't remove the old code to delete functions, just added it to runRegionalDeployment. Am I missing something?

**Comment by inlined** [comment]
Total oversight; should be fixed now.

File: src/deploy/functions/release.ts [hunk 1]
```diff
@@ -60,26 +60,21 @@ export async function release(context: args.Context, options: args.Options, payl
...
+  const allFnsToDelete = Object.values(fullDeployment.regionalDeployments).reduce(
+    (accum, region) => [...accum, ...region.functionsToDelete],
+    [] as backend.FunctionSpec[]
+  );
   const shouldDeleteFunctions = await promptForFunctionDeletion(
     allFnsToDelete,
     options.force,
     options.nonInteractive
   );
-  if (shouldDeleteFunctions) {
```
Comment (joehan) [inline src/deploy/functions/release.ts:71]: It looks like this new code will always delete functions, even shouldDeleteFunctons = false.

**Review by joehan — CHANGES_REQUESTED** [review]
I think this still may be broken - I don't see where you check whether the user wants to delet functions.

File: src/deploy/functions/release.ts (hunk 1 above)
Comment (inlined) [reply src/deploy/functions/release.ts:71]: @#$! it removes the topics and schedules from the delete list and still deletes the function. Will fix.

I guess the most dangerous PR is the one that's "too trivial to mess up"

**Review by joehan — APPROVED** [review]
LGTM now, as long as you've manually tested both cases

**Comment by inlined** [comment]
Tested both cases

**Pull request merged**